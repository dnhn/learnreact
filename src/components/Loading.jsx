import React from 'react';

import './Loading.scss';

export default ({ error }) =>
  <div className={`loading ${error ? 'error' : ''}`}>
    <span className="dot first" />
    <span className="dot second" />
    <span className="dot last" />
  </div>;
