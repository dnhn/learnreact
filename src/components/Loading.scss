.loading {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dot {
  display: inline-block;
  border-radius: 50%;
  background-color: white;
  width: 2vw;
  height: 2vw;
  margin: 1vw;
  animation-name: fade;
  animation-duration: .75s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  box-shadow: 0 2px 2px black;

  .error & {
    background-color: deeppink;
    animation-duration: 1.5s;
    animation-delay: 0s !important;
  }
}

.first {
  animation-delay: .5s;
}

.second {
  animation-delay: .75s;
}

.last {
  animation-delay: 1s;
}

@keyframes fade {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@media screen and (max-width: 640px) {
  .dot {
    width: 2vh;
    height: 2vh;
  }
}
