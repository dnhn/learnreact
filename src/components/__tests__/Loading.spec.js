import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../Loading';

it('renders correctly', () => {
  const { asFragment } = render(<Loading />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders error state', () => {
  const { container } = render(<Loading error />);
  expect(container.firstChild.classList.contains('error')).toBe(true);
});
