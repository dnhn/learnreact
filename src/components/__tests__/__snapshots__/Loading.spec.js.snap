// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading "
  >
    <span
      class="dot first"
    />
    <span
      class="dot second"
    />
    <span
      class="dot last"
    />
  </div>
</DocumentFragment>
`;
